src/pages/DisplayFixtures.jsx

```diff
@@ -99,11 +88,12 @@ const DisplayFixtures = () => {
 
   const handleSubmit = async (event) => {
     event.preventDefault();
+    setIsSaving(true);
 
     // const apiUrl = import.meta.env.VITE_API_URL;
     const apiUrl = import.meta.env.VITE_API_URL_USER_DB;
 
-    for (const prediction of predictions) {
```

Comment:
here, you were looping through `predictions` (which is essentially `userPredictions`) and calling `PUT` API for each loop. This means you'll be calling `PUT` API for each user prediction, regardless of whether that prediction has changed or not.

We could significantly improve this by tracking **just the predictions that have changed** (hence why I created the `changedPredictions` state variable, and keeping track of changed predictions inside `handleUpdatePrediction` handler on L39 & L40 above)

This way, we only call `PUT` API for predictions that have changed and eliminate all other unnecessary `PUT` API calls for predictions that didn't change!